shared/chat/conversation/messages/system-joined/container.tsx

```diff
@@ -12,7 +12,7 @@ const mapStateToProps = (state, {message}) => ({
   _meta: Constants.getMeta(state, message.conversationIDKey),
   author: message.author,
   authorIsYou: state.config.username === message.author,
-  joiners: message.joiners,
+  joiners: !message.joiners.length && !message.leavers.length ? [message.author] : message.joiners,
```

Comment:
why is this necessary? why doesn't the service do this?

Reply:
The problem is if you are looking at the thread, it doesn't go through the whole consolidation path.

Reply:
oh so this was preexisting?